src/index.ts

```diff
@@ -149,14 +154,29 @@ document.addEventListener('DOMContentLoaded', () => {
     });
   });
 
-  creditcardformSubmit?.addEventListener('click', function (e) {
-    e.preventDefault();
+  creditcardform?.addEventListener(
+    'submit',
+    async function (e) {
+      e.preventDefault();
+      const result = await pmClient.startSessionUsingPOST({
+        startSessionRequest: {
+          data: {
+            email: 'pippo@pluto.com',
+            fiscalCode: 'HBBJUU78U89R556T',
+            idPayment: '12345',
+          },
+        },
+      });
 
-    creditcardformInputs?.forEach(el => {
-      sessionStorage.setItem(el.getAttribute('name')?.trim() || '', el.value);
-    });
-    window.location.replace('check.html');
-  });
+      if (result.isRight()) {
```

Comment:
According to [this](https://pagopa.github.io/io-docs/io-handbook/development-styleguide), you can use _fold_ instead _isRight_.
Something like that:

```ts
      (
        await pmClient.startSessionUsingPOST({
          startSessionRequest: {
            data: {
              email: 'pippo@pluto.com',
              fiscalCode: 'HBBJUU78U89R556T',
              idPayment: '12345',
            },
          },
        })
      ).fold(
        () => console.log('some error'),  // to be replaced with logic to handle failures
        () => {
          creditcardformInputs?.forEach(el => {
            sessionStorage.setItem(el.getAttribute('name')?.trim() || '', el.value);
          });
          window.location.replace('check.html');
        },
      );
```

Reply:
@infantesimone Thank you for the snippet! I'll use it